cluster/external/provider.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+function _kubectl() {
+    kubectl "$@"
+}
+
+# The external cluster is assumed to be up.  Do a simple check
+function up() {
+    _kubectl version >/dev/null
+    if [ $? -ne 0 ]; then
+        echo -e "\n*** Unable to reach external cluster.  Please check configuration ***"
+        echo -e "*** Type \"kubectl config view\" for current settings               ***\n"
+        exit 1
+    fi
+    echo "Cluster is up"
+}
+
+function down() {
+    echo "Not supported by this provider"
+}
+
+function build() {
+    # Build code and manifests
+    make all
```

Comment:
I don't understand the difference, since the 'all' rule essentially does the same thing (build code and manifests), and abstracting it into a make rule seems cleaner, although I'm not a big fan of how make->script->make flow but that's neither here nor there.
I'm not fully convinced the push belongs here, but since anybody using this workflow will probably have an external registry, pushing will likely always follow a build.

Reply:
Actually it needs to look like this:

```
${KUBEVIRT_PATH}hack/dockerized "DOCKER_TAG=${DOCKER_TAG} KUBEVIRT_PROVIDER=${KUBEVIRT_PROVIDER} ./hack/build-manifests.sh"
make push
```

Only the manifest build step needs this extra step to pass in the provider.

The background is soley that. All other providers do the same thing right now.
The push needs to be part of it too, since this method is only called when you invoke `make cluster-sync`. It's purpose is to bring the cluster and the changed code at  the same state.

Reply:
OK, if the PR https://github.com/kubevirt/kubevirt/pull/1352 is dead in the water, I see why this is necessary.  If we can get it in soon then simple make calls here will suffice.